### PR TITLE
#902 Making bottom panel resizable

### PR DESCRIPTION
**Pull request by nabilIbm**
Fixes: #902 
Developer's Certificate of Origin 1.1

       By making a contribution to this project, I certify that:

       (a) The contribution was created in whole or in part by me and I
           have the right to submit it under the Apache License 2.0; or

       (b) The contribution is based upon previous work that, to the best
           of my knowledge, is covered under an appropriate open source
           license and I have the right under that license to submit that
           work with modifications, whether created in whole or in part
           by me, under the same open source license (unless I am
           permitted to submit under a different license), as indicated
           in the file; or

       (c) The contribution was provided directly to me by some other
           person who certified (a), (b) or (c) and I have not modified
           it.

       (d) I understand and agree that this project and the contribution
           are public and that a record of the contribution (including all
           personal information I submit with it, including my sign-off) is
           maintained indefinitely and may be redistributed consistent with
           this project or the open source license(s) involved.

### CODE REVIEW

**Review by tomlyn — CHANGES_REQUESTED** [review]
When I try this the panel jumps from where he mouse pointer position is localed
Bottom Panel

**Review by tomlyn — CHANGES_REQUESTED** [review]

File: canvas_modules/harness/cypress/support/index.js [hunk 1]
```diff
@@ -28,6 +28,7 @@ import "./canvas/toolbar-cmds";
 import "./canvas/utils-cmds";
 import "./canvas/verification-cmds";
 import "./canvas/operation-cmds";
+import "../integration/canvas/bottom-panel";
```
Comment (tomlyn) [inline canvas_modules/harness/cypress/support/index.js:31]: Why is this here? Do we need it? We don't list any of the other 'integration' files here which ought to be a clue that bottom-panel is not needed here.  This seems to be causing the testcase to run for a second time. Are you not seeing that when you run the bottom-panel test in Cypress?

File: canvas_modules/harness/cypress/support/canvas/verification-cmds.js [hunk 2]
```diff
@@ -746,6 +746,12 @@ Cypress.Commands.add("verifyNodeDimensions", (nodeId, width, height) => {
 		});
 });
 
+Cypress.Commands.add("verifyBoottomPanelHeight", (height) => {
```
Comment (tomlyn) [inline canvas_modules/harness/cypress/support/canvas/verification-cmds.js:749]: Typo verifyBoottomPanelHeight should be verifyBottomPanelHeight.

File: canvas_modules/harness/cypress/support/canvas/operation-cmds.js [hunk 3]
```diff
@@ -38,3 +38,15 @@ Cypress.Commands.add("selectWithRegion", (x1, y1, x2, y2) => {
 	});
 
 });
+
+Cypress.Commands.add("moveBottomPanelDivider", (yOffset) => {
+// movementY needs to be passed in , because within cc-bottom-panel.jsx onMouseMoveY() is checking
+// on mouse movementY amount which is not passed by Cypress.
```
Comment (tomlyn) [inline canvas_modules/harness/cypress/support/canvas/operation-cmds.js:44]: These comments should be indented the same amount as the code below it.

File: canvas_modules/harness/cypress/integration/canvas/bottom-panel.js [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+describe("Testing bottom panel", function() {
+	beforeEach(() => {
+		cy.visit("/");
+		cy.openCanvasDefinition("commentColorCanvas.json");
+	});
+
+	it("Testing bottom pannel seizing", function() {
```
Comment (tomlyn) [inline canvas_modules/harness/cypress/integration/canvas/bottom-panel.js:22]: "pannel seizing"  should be "panel sizing"

File: canvas_modules/harness/cypress/integration/canvas/bottom-panel.js [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+	// Open right side menu & toggle bottomPanel
```
Comment (tomlyn) [inline canvas_modules/harness/cypress/integration/canvas/bottom-panel.js:23]: This comment should be indented the same amount as the code below it. Also, the comment is wrong because the code is not opening the right side menu and toggling bottom panel. You can probably remove this comment.

**Review by tomlyn — CHANGES_REQUESTED** [review]

File: canvas_modules/common-canvas/src/common-canvas/common-canvas.scss [hunk 6]
```diff
@@ -199,5 +183,44 @@ $bottom-panel-height: 393px;
...
+	}
+}
 
+.bottom-panel-contents {
+	background-color: #FFFFFF;
+	flex: 1;
+	width: 1510px;
+	margin: auto;
+	user-select: none;
+}
```
[the remark this answers is not in the document]
Comment (matthoward366) [reply canvas_modules/common-canvas/src/common-canvas/common-canvas.scss:218]: I don't know why `background-color: #FFFFFF;` is set.  This should be up to the host application to set.  Why is `width` hardcoded?

[the remark this answers is not in the document]
Comment (tomlyn) [reply canvas_modules/common-canvas/src/common-canvas/common-canvas.scss:243]: Good catch @matthoward366   We set the default background-color of the palette and right-flyout panel to `$ui-01` , which the host app can override if it wishes, so I suggest you, Nabil, change it to that.

As for the width, that is working OK in the current code in master where the width of the bottom panel will automatically adjust depending on the size of the palette and size of the right-side flyout (if present). We need to make it so that behavior remains.

**Review by tomlyn — CHANGES_REQUESTED** [review]
Can you do the following please:
1. Remove the code that tries to position the graphic in the center of the canvas area
2. Change the text case in context-menu.js back to the way it was before and change this test case so it is skipped -- that is, instead of saying
```
it("Test context menu position is moved correctly with bottom panl open", function() {
```
it should say 
```
it.skip("Test context menu position is moved correctly with bottom panl open", function() {
```
3. In the text for that test case change `panl` to `panel`
4. The comments in operation-cmds.js need to be indented. I mentiond that in my previous review comment but you seem to have ignored it.

File: canvas_modules/harness/cypress/integration/canvas/context-menu.js [hunk 7]
```diff
@@ -88,7 +88,7 @@ describe("Test of context menu", function() {
 		cy.verifyNumberOfSelectedObjects(0);
 	});
 
-	it("Test context menu position is moved correctly with bottom panl open", function() {
+	it.skip("Test context menu position is moved correctly with bottom panl open", function() {
```
Comment (tomlyn) [inline canvas_modules/harness/cypress/integration/canvas/context-menu.js:91]: Please add a comment here to indicate why we are skipping this test. Say something like: 
"This test is being skipped because the canvas area needs to be fixed. At the moment it stretches underneath the bottom panel to the bottom of the common canvas div. It needs to be fixed so that, when the bottom panel is open, the canvas area only extends down to where the bottom panel begins."

File: canvas_modules/common-canvas/src/common-canvas/common-canvas.scss [hunk 8]
```diff
@@ -88,15 +83,34 @@ $bottom-panel-height: 393px;
...
+		background: $interactive-01;
+	}
+}
+
+.bottom-panel-contents {
+	background-color: $ui-01;
+	width: 100%;
+	user-select: none;
+	height: 100%;
+
```
Comment (tomlyn) [inline canvas_modules/common-canvas/src/common-canvas/common-canvas.scss:111]: Please add a min-height property of 75px.  (I'm not sure if this needs to be in this rule set or the .bottom-panel rule set so please test it out to make sure your new property works OK. It should  restricts the user from dragging the bottom panel to be less than 75px high)

**Review by tomlyn — CHANGES_REQUESTED** [review]

File: canvas_modules/common-canvas/src/common-canvas/common-canvas.scss (hunk 8 above)
Comment (tomlyn) [reply canvas_modules/common-canvas/src/common-canvas/common-canvas.scss:111]: Are you setting "min-height" ? In CSS "min-height" (which means minimum height) is different to "height".  You need to set it on `.bottom-panel` for it to take effect.

File: canvas_modules/harness/cypress/integration/canvas/context-menu.js (hunk 7 above)
Comment (tomlyn) [reply canvas_modules/harness/cypress/integration/canvas/context-menu.js:91]: You seem to have resolved this review comment without adding the comment I suggested to the code.

**Review by tomlyn — CHANGES_REQUESTED** [review]

**Review by tomlyn — APPROVED** [review]
